src/firestore/firestore.ts

```diff
@@ -14,20 +14,115 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 // =============================================================================
-import { Unsubscribe } from "firebase/firestore";
+import { initializeApp } from "firebase/app";
+import {
+  connectAuthEmulator,
+  getAuth,
+  signInWithCustomToken,
+} from "firebase/auth";
+import {
+  collection,
+  CollectionReference,
+  connectFirestoreEmulator,
+  getDocs,
+  getFirestore,
+  limit,
+  onSnapshot,
+  query,
+  Unsubscribe,
+  where,
+} from "firebase/firestore";
 
+import { TenantId } from "../RootStore/types";
 import {
   ClientRecord,
   ClientUpdateRecord,
   CombinedUserRecord,
   OpportunityType,
-  UserRecord,
+  StaffRecord,
+  UserUpdateRecord,
 } from "./types";
 
+const app = initializeApp({
+  projectId: process.env.REACT_APP_FIRESTORE_PROJECT,
+  apiKey: "fakevaluefornow",
+});
+
+export const authenticate = async (
+  auth0Token: string
+): Promise<ReturnType<typeof signInWithCustomToken>> => {
+  const tokenExchangeResponse = await fetch(
+    `${process.env.REACT_APP_API_URL}/token`,
+    {
+      headers: {
+        Authorization: `Bearer ${auth0Token}`,
+      },
+    }
+  );
+
+  const { firebaseToken } = await tokenExchangeResponse.json();
+  const auth = getAuth(app);
+  if (process.env.REACT_APP_FIRESTORE_PROJECT?.startsWith("demo-")) {
+    connectAuthEmulator(auth, "http://localhost:9099");
+  }
+  return signInWithCustomToken(auth, firebaseToken);
+};
+
+const db = getFirestore(app);
+if (process.env.REACT_APP_FIRESTORE_PROJECT?.startsWith("demo-")) {
+  connectFirestoreEmulator(db, "localhost", 8080);
+}
+const collections = {
+  staff: collection(db, "staff") as CollectionReference<StaffRecord>,
+  userUpdates: collection(
+    db,
+    "userUpdates"
+  ) as CollectionReference<UserUpdateRecord>,
+  clients: collection(db, "clients") as CollectionReference<ClientRecord>,
+  clientUpdates: collection(
+    db,
+    "clientUpdates"
+  ) as CollectionReference<ClientUpdateRecord>,
+};
+
 export async function getUser(
-  email: string
+  email: string,
+  stateCode: TenantId
```

Comment:
nit: trailing comma for easier future editing
```suggestion
  stateCode: TenantId,
```

Reply:
prettier does not condone this